prov/psm2: Compilation warnings

The following warnings are produces when `./configure --enable-psm2 ...`

```
prov/psm2/src/psmx2_atomic.c: In function ‘psmx2_atomic_write_generic’:
prov/psm2/src/psmx2_atomic.c:847: warning: comparison of unsigned expression < 0 is always false
prov/psm2/src/psmx2_atomic.c:850: warning: comparison of unsigned expression < 0 is always false
prov/psm2/src/psmx2_atomic.c: In function ‘psmx2_atomic_writev_generic’:
prov/psm2/src/psmx2_atomic.c:987: warning: comparison of unsigned expression < 0 is always false
prov/psm2/src/psmx2_atomic.c:990: warning: comparison of unsigned expression < 0 is always false
prov/psm2/src/psmx2_atomic.c: In function ‘psmx2_atomic_readwrite_generic’:
prov/psm2/src/psmx2_atomic.c:1222: warning: comparison of unsigned expression < 0 is always false
prov/psm2/src/psmx2_atomic.c:1225: warning: comparison of unsigned expression < 0 is always false
prov/psm2/src/psmx2_atomic.c: In function ‘psmx2_atomic_readwritev_generic’:
prov/psm2/src/psmx2_atomic.c:1379: warning: comparison of unsigned expression < 0 is always false
prov/psm2/src/psmx2_atomic.c:1382: warning: comparison of unsigned expression < 0 is always false
prov/psm2/src/psmx2_atomic.c: In function ‘psmx2_atomic_compwrite_generic’:
prov/psm2/src/psmx2_atomic.c:1693: warning: comparison of unsigned expression < 0 is always false
prov/psm2/src/psmx2_atomic.c:1696: warning: comparison of unsigned expression < 0 is always false
prov/psm2/src/psmx2_atomic.c: In function ‘psmx2_atomic_compwritev_generic’:
prov/psm2/src/psmx2_atomic.c:1858: warning: comparison of unsigned expression < 0 is always false
prov/psm2/src/psmx2_atomic.c:1861: warning: comparison of unsigned expression < 0 is always false
```

The following warning are produces when `./configure --enable-psm2=dl ...`

```
prov/psm2/src/psmx2_rma.c: In function ‘psmx2_readv_generic’:
prov/psm2/src/psmx2_rma.c:752: warning: ‘long_buf’ may be used uninitialized in this function
```
This warning is false positive, because the `long_len` variable controls the branch where `long_buf` may be uninitialized.

Not sure why it's produced only for these options of the configure

--- a/prov/psm2/src/psmx2_atomic.c
+++ b/prov/psm2/src/psmx2_atomic.c
@@ -844,10 +844,10 @@ ssize_t psmx2_atomic_write_generic(struct fid_ep *ep,
 	if (!buf)
 		return -FI_EINVAL;
 
-	if (datatype < 0 || datatype >= FI_DATATYPE_LAST)
+	if ((int)datatype < 0 || (int)datatype >= FI_DATATYPE_LAST)
 		return -FI_EINVAL;
 
-	if (op < 0 || op >= FI_ATOMIC_OP_LAST)
+	if ((int)op < 0 || (int)op >= FI_ATOMIC_OP_LAST)
 		return -FI_EINVAL;
 
 	av = ep_priv->av;
@@ -984,10 +984,10 @@ ssize_t psmx2_atomic_writev_generic(struct fid_ep *ep,
 	while (count && !iov[count-1].count)
 		count--;
 
-	if (datatype < 0 || datatype >= FI_DATATYPE_LAST)
+	if ((int)datatype < 0 || (int)datatype >= FI_DATATYPE_LAST)
 		return -FI_EINVAL;
 
-	if (op < 0 || op >= FI_ATOMIC_OP_LAST)
+	if ((int)op < 0 || (int)op >= FI_ATOMIC_OP_LAST)
 		return -FI_EINVAL;
 
 	av = ep_priv->av;
@@ -1219,10 +1219,10 @@ ssize_t psmx2_atomic_readwrite_generic(struct fid_ep *ep,
 	if (!buf && op != FI_ATOMIC_READ)
 		return -FI_EINVAL;
 
-	if (datatype < 0 || datatype >= FI_DATATYPE_LAST)
+	if ((int)datatype < 0 || (int)datatype >= FI_DATATYPE_LAST)
 		return -FI_EINVAL;
 
-	if (op < 0 || op >= FI_ATOMIC_OP_LAST)
+	if ((int)op < 0 || (int)op >= FI_ATOMIC_OP_LAST)
 		return -FI_EINVAL;
 
 	av = ep_priv->av;
@@ -1376,10 +1376,10 @@ ssize_t psmx2_atomic_readwritev_generic(struct fid_ep *ep,
 	while (result_count && !resultv[result_count-1].count)
 		result_count--;
 
-	if (datatype < 0 || datatype >= FI_DATATYPE_LAST)
+	if ((int)datatype < 0 || (int)datatype >= FI_DATATYPE_LAST)
 		return -FI_EINVAL;
 
-	if (op < 0 || op >= FI_ATOMIC_OP_LAST)
+	if ((int)op < 0 || (int)op >= FI_ATOMIC_OP_LAST)
 		return -FI_EINVAL;
 
 	result_len = psmx2_ioc_size(resultv, result_count, datatype);
@@ -1690,10 +1690,10 @@ ssize_t psmx2_atomic_compwrite_generic(struct fid_ep *ep,
 	if (!buf)
 		return -FI_EINVAL;
 
-	if (datatype < 0 || datatype >= FI_DATATYPE_LAST)
+	if ((int)datatype < 0 || (int)datatype >= FI_DATATYPE_LAST)
 		return -FI_EINVAL;
 
-	if (op < 0 || op >= FI_ATOMIC_OP_LAST)
+	if ((int)op < 0 || (int)op >= FI_ATOMIC_OP_LAST)
 		return -FI_EINVAL;
 
 	av = ep_priv->av;
@@ -1855,10 +1855,10 @@ ssize_t psmx2_atomic_compwritev_generic(struct fid_ep *ep,
 	while (result_count && !resultv[result_count-1].count)
 		result_count--;
 
-	if (datatype < 0 || datatype >= FI_DATATYPE_LAST)
+	if ((int)datatype < 0 || (int)datatype >= FI_DATATYPE_LAST)
 		return -FI_EINVAL;
 
-	if (op < 0 || op >= FI_ATOMIC_OP_LAST)
+	if ((int)op < 0 || (int)op >= FI_ATOMIC_OP_LAST)
 		return -FI_EINVAL;
 
 	len = psmx2_ioc_size(iov, count, datatype);

--- a/prov/psm2/src/psmx2_rma.c
+++ b/prov/psm2/src/psmx2_rma.c
@@ -748,8 +748,8 @@ ssize_t psmx2_readv_generic(struct fid_ep *ep, const struct iovec *iov,
 	psm2_mq_tag_t psm2_tag, psm2_tagsel;
 	uint32_t tag32;
 	size_t idx;
-	size_t total_len, long_len, short_len;
-	void *long_buf;
+	size_t total_len, long_len = 0, short_len;
+	void *long_buf = NULL;
 	int i;
 	int err;
 
@@ -843,7 +843,6 @@ ssize_t psmx2_readv_generic(struct fid_ep *ep, const struct iovec *iov,
 
 	chunk_size = ep_priv->trx_ctxt->psm2_am_param.max_reply_short;
 
-	long_len = 0;
 	if (psmx2_env.tagged_rma) {
 		for (i=count-1; i>=0; i--) {
 			if (iov[i].iov_len > chunk_size) {

